signer/hsm.go

```diff
@@ -107,6 +107,36 @@ type GCPHSM struct {
 	GenericHSM
 }
 
+type unlimitedBytesRandReader struct {
+	rr crypto11.PKCS11RandReader
+}
+
+func (ubrr *unlimitedBytesRandReader) Read(data []byte) (n int, err error) {
+	var result []byte
+	need := len(data)
+	for need > 0 {
+		var bytesRead int
+		next := make([]byte, min(need, 1024))
+		bytesRead, err = ubrr.rr.Read(next)
+		n += bytesRead
+		if err != nil {
+			return
```

Comment:
Is there any way to not lose the error here and return/log it?

Reply:
AIUI, we're returning that error (the `err` variable is bound to the return as a "named result parameter": https://go.dev/doc/effective_go#named-results)

I think this is verified by the new test where we mock in a specific error that `GenerateRandom` returns, and verify that it comes out of UnlimitedBytesRandReader.Read: https://github.com/mozilla-services/autograph/pull/1032/files#diff-a43f626c786b027528255923aee643b55513fb5d596a69e082c63dba5b9ccfb9R833-R857

Reply:
Ah, overlooked that thank you.

Reply:
Sure. I thought they were idiomatic, but I have no real preference.

Reply:
Some folks in Go-land like them, but I think the community is mostly on the side of "too easy to have shadowing problems".